sympy/integrals/transforms.py

```diff
@@ -1777,8 +1773,13 @@ def _laplace_apply_rules(f, t, s, doit=True, **hints):
     prog_rules = [_laplace_rule_timescale, _laplace_rule_heaviside,
                   _laplace_rule_exp, _laplace_rule_trig, _laplace_rule_diff]
     for p_rule in prog_rules:
-        LT = p_rule(f, t, s, doit=doit, **hints)
+        LT = p_rule(func, t, s, doit=doit, **hints)
         if LT is not None:
+            try:
+                r, p, c = LT
+                return (k*r, p, c)
+            except TypeError:
+                return k*LT
```

Comment:
Rather than `try/except` it would be better to test the condition explicitly e.g.:
```python
conds = hints.get('noconds', False):
if conds:
    r, p, c = LT
    return (k*r, p, c)
else:
    return k*LT
```
This is better because:
- It is more explicit
- We should generally avoid catching TypeError which is usually indicative of bugs
- Some expressions are iterable so there is a possibility of confusion here:
   ```python
   In [1]: r, p, c = Matrix([1, 2, 3])

   In [2]: r, p, c
   Out[2]: (1, 2, 3)
   ```
- This also ensures that when `noconds` is set we will see an error if the wrong type of object was returned here.

In general catching exceptions is something that should be used for exceptional rather than expected cases.

Reply:
That was how we handled it a year ago, but I agree it is not good. I have rewritten all instances. However, since it is not always certain inside the recursive rule engine that conditions are returned even if `nocond` is not there, I am now also checking whether the return value's type is tuple.

Push will follow soon ...

Reply:
> it is not always certain inside the recursive rule engine that conditions are returned even if `nocond` is not there

This is precisely why I would check `nocond`: if `nocond` is passed and a tuple is being returned then that's a bug is it not? Surely in that case we should *want* to see a `TypeError` rather than silently passing the wrong type through.

Reply:
It is the other way around: if noconds is not passed, it can still be that no tuple is returned inside the recursive tree.

Reply:
Okay that makes sense. I think it's not a good API to return either a tuple or a LaplaceTransform but I guess that can't be changed now.